---
title: "Exercices in linear models and experimental design"
author: "Ana"
date: "30/5/2019"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---

##Introduction
These exercises describe experimental situations that have been studied using some omics technology like microarrays.

##Expression profiling by array

This case study is based on a paer published in  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE113664 whose data are available in GEO as series GSE113664 series.

UHRF1 is a major regulator of epigenetic mechanism and is overexpressed in various human malignancies. In this study, we examined the involvement of UHRF1 in aberrant DNA methylation in colorectal cancer (CRC). In CRC cells, transient UHRF1 knockdown rapidly induced DNA demethylation across entire genomic regions, including CpG islands, gene bodies and repetitive elements. Nonetheless, UHRF1 depletion only minimally reversed CpG island hypermethylation-associated gene silencing. However, the combination of UHRF1 depletion and histone deacetylase (HDAC) inhibition synergistically reactivated the silenced genes and strongly suppressed CRC cell proliferation. Our results suggest that (i) maintenance of DNA methylation in CRC cells is highly dependent on UHRF1; (ii) UHRF1 depletion rapidly induces DNA demethylation, though it is insufficient to fully reactivate the silenced genes; and (iii) dual targeting of UHRF1 and HDAC may be an effective new therapeutic strategy.


```{r readTargets}
require (limma)
targetsLocation<- 'sample.txt'
targets <- read.csv('sample.txt', sep = '')

targets
```

```{r readData}
dataLocation<- 'GSE113664_series_matrix.txt'
dataMatrix <- read.delim("GSE113664_series_matrix.txt", row.names = 1)
colnames(dataMatrix)==rownames(targets)
head(dataMatrix)

```

```{r designMatrix}
design<-matrix(
  c(1,1,0,0,0,0,0,0,0,
    0,0,1,1,0,0,0,0,0,
    0,0,0,0,1,1,0,0,0,
    0,0,0,0,0,0,1,1,0,
    0,0,0,0,0,0,0,0,1),
  nrow=9,byrow=F)
design2 <-model.matrix(~ 0+targets$Group)
colnames(design)<- colnames(design2)<-c("A", "B", "C" , "D", "E")
rownames(design) <- rownames(design2)<- targets$SampleIDs 
print(design); print(design2)
```


```{r contrastsMatrix}
cont.matrix <- makeContrasts (
  AvsB = B-A,
  AvsC = C-A,
  AvsD = D-B,
  AvsE = E-A,
  BvsC = C-B,
  BvsD = D-B,
  BvsE = E-B,
  CvsD = D-C,
  CvsE = E-C,
  DvsE = E-D,
  levels=design)
design
```

```{r fitModel}
require(limma)
fit<-lmFit(dataMatrix, design)
fit.main<-contrasts.fit(fit, cont.matrix)
fit.main<-eBayes(fit.main)
```

```{r extractResults}
topTab_AvsB <- topTable (fit.main, number=nrow(fit.main), coef="AvsB", adjust="fdr"); head(topTab_AvsB)
topTab_AvsC <- topTable (fit.main, number=nrow(fit.main), coef="AvsC", adjust="fdr"); head(topTab_AvsC)
topTab_AvsD  <- topTable (fit.main, number=nrow(fit.main) , coef="AvsD", adjust="fdr");head(topTab_AvsD)
topTab_AvsE  <- topTable (fit.main, number=nrow(fit.main) , coef="AvsE", adjust="fdr");head(topTab_AvsE)
topTab_BvsC <- topTable (fit.main, number=nrow(fit.main) , coef="BvsC", adjust="fdr"); head(topTab_BvsC)
topTab_BvsE <- topTable (fit.main, number=nrow(fit.main) , coef="BvsE", adjust="fdr"); head(topTab_BvsE)
topTab_BvsD <- topTable (fit.main, number=nrow(fit.main) , coef="BvsD", adjust="fdr"); head(topTab_BvsD)
topTab_CvsD <- topTable (fit.main, number=nrow(fit.main) , coef="BvsC", adjust="fdr"); head(topTab_CvsD)
topTab_CvsE <- topTable (fit.main, number=nrow(fit.main) , coef="CvsE", adjust="fdr"); head(topTab_CvsE)
topTab_DvsE <- topTable (fit.main, number=nrow(fit.main) , coef="DvsE", adjust="fdr"); head(topTab_DvsE)



```

```{r showResults}
volcanoplot(fit.main, coef="AvsB", highlight=10)
volcanoplot(fit.main, coef="AvsC", highlight=10)
volcanoplot(fit.main, coef="AvsD", highlight=10)
volcanoplot(fit.main, coef="AvsE", highlight=10)
volcanoplot(fit.main, coef="BvsC", highlight=10)
volcanoplot(fit.main, coef="BvsD", highlight=10)
volcanoplot(fit.main, coef="BvsE", highlight=10)
volcanoplot(fit.main, coef="CvsD", highlight=10)
volcanoplot(fit.main, coef="CvsE", highlight=10)
volcanoplot(fit.main, coef="DvsE", highlight=10)


```

```{r}
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE27174")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]
```






