---
title: "OMICS"
author: "Ana"
date: "22/5/2019"
output:
  html_document: default
  pdf_document: default
editor_options:
  chunk_output_type: console
---
##Summary
UHRF1 is a major regulator of epigenetic mechanism and is overexpressed in various human malignancies. In this study, we examined the involvement of UHRF1 in aberrant DNA methylation in colorectal cancer (CRC). In CRC cells, transient UHRF1 knockdown rapidly induced DNA demethylation across entire genomic regions, including CpG islands, gene bodies and repetitive elements. Nonetheless, UHRF1 depletion only minimally reversed CpG island hypermethylation-associated gene silencing. However, the combination of UHRF1 depletion and histone deacetylase (HDAC) inhibition synergistically reactivated the silenced genes and strongly suppressed CRC cell proliferation. Our results suggest that (i) maintenance of DNA methylation in CRC cells is highly dependent on UHRF1; (ii) UHRF1 depletion rapidly induces DNA demethylation, though it is insufficient to fully reactivate the silenced genes; and (iii) dual targeting of UHRF1 and HDAC may be an effective new therapeutic strategy.

##Overall design	
RKO cells were transfected with a control siRNA (Thermo Fisher Scientific) or siRNAs targeting UHRF1 (Thermo Fisher Scientific) using using Lipofectamine RNAiMAX (Thermo Fisher Scientific) and incubated for 72 h. Cells were then treated with mock (ethanol) or 300 nM Trichostatin A (TSA) for additional 24 h. RKO cells were also treated with AZA.

##Contributor(s)	
Suzuki H

##Citation(s)	
Niinuma T, Kitajima H, Kai M, Yamamoto E et al. UHRF1 depletion and HDAC inhibition reactivate epigenetically silenced genes in colorectal cancer cells. Clin Epigenetics 2019 May 7;11(1):70. PMID: 31064417

```{r}
targets <- read.csv('sample.txt', sep = '')
targets
```
Here, we are reading our samples file
```{r}
expressions <- read.delim('GSE113664_series_matrix.txt' , header = TRUE, row.names = 1)
expressions
```

```{r}
dim(expressions)
summary(expressions)

```

```{r}
boxplot(expressions, col = c(5,5,7,7,8,8,6,6,9))

```
This output is a boxplot of our data. We can't see a clear box because the data is dispersed and not localized. There's not a lot of  difference between all the expressions, they have some outliers. Bu, despite that we can conclude that they are regular.

```{r}
boxplot(log(expressions), col = c(5,5,7,7,8,8,6,6,9))

```
When we normalize the data, we get the boxes. They are very similar in all the groups despite the last one that seems to have higher values than the others. It's not a strange fact because if we look to our data, it seems that GSM3118962(last expression) is the one that is more different than the others.

```{r}
## ----simulateData--------------------------------------------------------
expressionValues <- data.matrix(expressions)
head(expressionValues)


```
Here, we are converting our data frame into a matrix


```{r}
## ----simulateGeneInfo----------------------------------------------------
myGenes<-rownames(expressions)
myGenes
```
We define the identifiers.


```{r}
## ----simulateInfo--------------------------------------------------------
myInfo=list(myName="Ana Sanchez", myLab="Bioinformatics Lab", 
          myContact="ana.sanchez@alum.esci.upf.edu", myTitle="Practical Exercise on ExpressionSets")
show(myInfo)
```


```{r}
## ------------------------------------------------------------------------
variab <- apply(expressionValues, 1, sd)
orderedGenes <- myGenes[order(variab, decreasing=TRUE)]
head(variab[order(variab, decreasing=TRUE)])
head(orderedGenes)

```

```{r}
## ----subsetExpressions---------------------------------------------------
newExpress<- expressionValues[,-9]
newTargets <- targets[-9,]
wrongNewTargets <- targets [-10,]
newExpress
newTargets
wrongNewTargets
```

```{r}
require("Biobase")
```

```{r}
## ----creaExpressionSet1--------------------------------------------------
myEset <- ExpressionSet(expressionValues)
class(myEset)
show(myEset)

```
We're using ExpressionSet in order to initiate an object or a class from our dataset

```{r}

## ----AnnotatedDataFrame2-------------------------------------------------
columnDesc <-  data.frame(labelDescription= c("Identifier", "Type", "Drug"))
myAnnotDF <- new("AnnotatedDataFrame", data=targets, varMetadata= columnDesc)
show(myAnnotDF)

```
By using data.frame, we are adding information to the columns.
```{r}
## ------------------------------------------------------------------------
phenoData(myEset) <- myAnnotDF 
phenoData(myEset)
```
We are storing the information in the slot phenoData.

```{r}

## ------------------------------------------------------------------------
rownames(pData(myAnnotDF))<-(pData(myAnnotDF)$Identifier)
pData(myAnnotDF)==colnames(expressionValues)
myEset <- ExpressionSet(assayData=expressionValues, phenoData=myAnnotDF)
show(myEset)

```

```{r}

## ------------------------------------------------------------------------
myEset <- ExpressionSet(assayData=expressionValues, 
                        phenoData=myAnnotDF, 
                        featureNames =myGenes)
# show(myEset)

```
Then, we can do this store the information about our identifiers(genes) using a character vector.

```{r}
## ----label=MIAME---------------------------------------------------------
myDesc <- new("MIAME", name= myInfo[["myName"]],
            lab= myInfo[["myLab"]],
            contact= myInfo[["myContact"]] ,
            title=myInfo[["myTitle"]])
print(myDesc)

```
Another way to store information about the information is by using "MIAME". Not necessary for the data analysis.

```{r}
## ------------------------------------------------------------------------
myEset <- ExpressionSet(assayData=expressionValues, 
                        phenoData=myAnnotDF,
                        fetureNames =myGenes,
                        experimentData = myDesc)
# show(myEset)
```
Then we add another object to our Expression set.
```{r}
## ----usingExpressionSets-------------------------------------------------
dim(exprs(myEset))
class(phenoData(myEset))
class(pData(phenoData(myEset)))
head(pData(phenoData(myEset)))
head(pData(myEset))

```
This is used for accessing slot values.

```{r}

## ------------------------------------------------------------------------
smallEset <- myEset[1:15,c(1:3,6:8)]
dim(exprs(smallEset))
dim(pData(smallEset))
head(pData(smallEset))
all(colnames(exprs(smallEset))==rownames(pData(smallEset)))
smallEset

```
This is done for subsetting the ExpressionSet.
For example, I have made reduced dataset that contains 6 samples.


```{r}
## ------------------------------------------------------------------------
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE27174")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]
```













