###ABSTRACT
In this study, it has been examined the involvement of UHRF1, a major regulator of epigenetic mechanism, in abnormal DNA methylation in colorectal cancer (CRC). 

In CRC cells, THE temporary UHRF1 AUGMENT induced DNA demethylation across entire genomic regions, including CpG islands, gene bodies and repetitive elements.
UHRF1 REDUCTION MAKE A only minimalL EFFECT ON GENE SILENCING.
BUT IF WE REDUCE THE UHRF1 AND ALSO INHIBIT HISTONE DEACETYLASE (HDAC) there's a strong supression oon CRC cell proliferation by the silenced genes. 
###OBJECTIVES
In this experiment we want to know the effect of UHRF1 in CRC cells.

###MATERIALS AND METHODS
###Data type, type of experiment,experimental design,
Data type that we have u sed is about Targets and GSE data.It is an non-experimental experiment and it’s done this way:First we collect the data and we make plots about this row data, then, we normalize it and obtainother plots.We want to compare the the different the raw and the normalized data and we can do so by comparingthe different plots.
### Methods that have been used in the analysis
###General procedure of analysis (steps, “workflow” or “pipeline” that you
RKO cells were transfected with a control siRNA (Thermo Fisher Scientific) or siRNAs targeting UHRF1 (Thermo Fisher Scientific) using using Lipofectamine RNAiMAX (Thermo Fisher Scientific) and incubated for 72 h. Cells were then treated with mock (ethanol) or 300 nM Trichostatin A (TSA) for additional 24 h. Alternatively, RKO cells were treated with 1 microM 5-aza-2'-deoxycytidine for 72 h.
### Procedures applied in each step:
Growth protocol:
RKO cells were cultured in DMEM supplemented with 10% heat-inactivated fetal bovine serum. The cultures were incubated at 37 ºC in a humidified incubator with 5% CO2.
Extraction protocol:
Total RNA was extracted using a TRIZOL reagent.
Label protocol: 
Total RNA (100 ng) was labeled using a Low Input Quick Amp Labeling Kit One-Color (Agilent Technologies) according to manufacturer's instruction.
Hybridization protocol:
Hybridization was performed using a Gene Expression Hybridization kit (Agilent Technologies) according to manufacturer's instruction.
Scan protocol:
Array was scanned with an Agilent G2565BA Microarray Scanner.


---------------------------------------------------------------------------------------------
###THIS IS AN EXAMPLE CODE FOR THE ANALYSIS OF AFFYMETRIX GENE MICROARRAYS
#---------------------------------------------------------------------------------------------


#---------------------------------------------------------------------------------------------
###FOLDER DESTINATION DEFINITIONS
#---------------------------------------------------------------------------------------------
```{r}
workingDir <-getwd()
dataDir <- file.path(workingDir, "sample.txt")
resultsDir <- file.path(workingDir, "GSE113664_series_matrix.txt")

```


#---------------------------------------------------------------------------------------------
###INSTALLATION OF PACKAGES NEEDED
#---------------------------------------------------------------------------------------------
```{r}

installifnot <- function (pkg){
  if (!require(pkg, character.only=T)){
    BiocManager::install(pkg)
  }else{
    require(pkg, character.only=T)
  }
}

installifnot("pd.mogene.1.0.st.v1")
installifnot("mogene10sttranscriptcluster.db")
installifnot("oligo")
installifnot("limma")
installifnot("Biobase")
installifnot("arrayQualityMetrics")
installifnot("genefilter")
installifnot("multtest")
installifnot("annotate")
installifnot("xtable")
installifnot("gplots")
installifnot("scatterplot3d")
BiocManager::install("hgu133a.db")
installifnot("stats")
installifnot("graphics")    
installifnot("grDevices")
installifnot("utils")
installifnot("datasets")
installifnot("methods")
installifnot("base")

```


#---------------------------------------------------------------------------------------------
###LOAD DATA: TARGETS AND CEL FILES.(NO CELL FILES: ESET). 
#---------------------------------------------------------------------------------------------

#TARGETS
```{r}
targets <-read.delim("targets.txt", header = TRUE, sep=";")
targets
```



#DATA
```{r}
expressions <-read.delim("~/Desktop/Sanchez_Ana_OMICSTECHNIQUES_Class/Last_Ex/GSE113664_series_matrix.txt", header = TRUE, row.names = 1)
expressions
rawData <-new("ExpressionSet",exprs=as.matrix(expressions))
head(rawData)
sampleNames <- as.character(targets$ShortName)
sampleNames
sampleColor <- as.character(targets$Color)
sampleColor
```



#---------------------------------------------------------------------------------------------
###QUALITY CONTROL OF ARRAYS: RAW DATA
#---------------------------------------------------------------------------------------------


#BOXPLOT
```{r}
boxplot(rawData, which="all",las=2, main="Intensity distribution of RAW data", 
        cex.axis=0.6, col=sampleColor, names=sampleNames)

```



#HIERARQUICAL CLUSTERING
```{r}
clust.euclid.average <- hclust(dist(t(exprs(rawData))),method="average")
plot(clust.euclid.average, labels=sampleNames, main="Hierarchical clustering of RawData", 
     cex=0.7,  hang=-1)
```



#PRINCIPAL COMPONENT ANALYSIS
```{r}
plotPCA <- function ( X, labels=NULL, colors=NULL, dataDesc="", scale=FALSE, formapunts=NULL, myCex=0.8,...)
{
  pcX<-prcomp(t(X), scale=scale) # o prcomp(t(X))
  loads<- round(pcX$sdev^2/sum(pcX$sdev^2)*100,1)
  xlab<-c(paste("PC1",loads[1],"%"))
  ylab<-c(paste("PC2",loads[2],"%"))
  if (is.null(colors)) colors=1
  plot(pcX$x[,1:2],xlab=xlab,ylab=ylab, col=colors, pch=formapunts, 
       xlim=c(min(pcX$x[,1])-100000, max(pcX$x[,1])+100000),ylim=c(min(pcX$x[,2])-100000, max(pcX$x[,2])+100000))
  text(pcX$x[,1],pcX$x[,2], labels, pos=3, cex=myCex)
  title(paste("Plot of first 2 PCs for expressions in", dataDesc, sep=" "), cex=0.8)
}

plotPCA(exprs(rawData), labels=sampleNames, dataDesc="raw data", colors=sampleColor,
        formapunts=c(rep(16,4),rep(17,4)), myCex=0.6)

```





#SAVE TO A FILE
```{r}
eset <- normalize(rawData)
eset

pdf(file.path(resultsDir, "QCPlots_Raw.pdf"))

boxplot(rawData, which="all",las=2, main="Intensity distribution of RAW data", 
        cex.axis=0.6, col=sampleColor, names=sampleNames)
plot(clust.euclid.average, labels=sampleNames, main="Hierarchical clustering of samples of RawData", 
     cex=0.7,  hang=-1)
plotPCA(exprs(rawData), labels=sampleNames, dataDesc="raw data", colors=sampleColor,
        formapunts=c(rep(16,4),rep(17,4)), myCex=0.6)


```




#---------------------------------------------------------------------------------------------
###DATA NORMALIZATION
#---------------------------------------------------------------------------------------------
```{r}
installifnot("affyPLM")
eset<-normalize.ExpressionSet.invariantset(rawData)
eset
write.exprs(eset,file.path("/Desktop/Sanchez_Ana_OMICSTECHNIQUES_Class/Last_Ex", "NormData.txt")

```



#---------------------------------------------------------------------------------------------
###QUALITY CONTROL OF ARRAYS: NORMALIZED DATA
#---------------------------------------------------------------------------------------------

#BOXPLOT
```{r}

boxplot(eset, which="all",las=2, main="Intensity distribution of Normal Data", 
        cex.axis=0.6, col=sampleColor, names=sampleNames)
e```


#HIERARQUICAL CLUSTERING
```{r}
clust.euclid.average <- hclust(dist(t(exprs(eset))),method="average")
plot(clust.euclid.average, labels=sampleNames, main="Hierarchical clustering of NormData", 
     cex=0.7,  hang=-1)
```


#PRINCIPAL COMPONENT ANALYSIS
```{r}
plotPCA <- function ( X, labels=NULL, colors=NULL, dataDesc="", scale=FALSE, formapunts=NULL, myCex=0.8,...)
{
  pcX<-prcomp(t(X), scale=scale) # o prcomp(t(X))
  loads<- round(pcX$sdev^2/sum(pcX$sdev^2)*100,1)
  xlab<-c(paste("PC1",loads[1],"%"))
  ylab<-c(paste("PC2",loads[2],"%"))
  if (is.null(colors)) colors=1
  plot(pcX$x[,1:2],xlab=xlab,ylab=ylab, col=colors, pch=formapunts, 
       xlim=c(min(pcX$x[,1])-10, max(pcX$x[,1])+10),ylim=c(min(pcX$x[,2])-10, max(pcX$x[,2])+10))
  text(pcX$x[,1],pcX$x[,2], labels, pos=3, cex=myCex)
  title(paste("Plot of first 2 PCs for expressions in", dataDesc, sep=" "), cex=0.8)
}
plotPCA(exprs(eset), labels=sampleNames, dataDesc="NormData", colors=sampleColor,
        formapunts=c(rep(16,4),rep(17,4)), myCex=0.6)
```



#SAVE TO A FILE
```{r}
boxplot(eset, las=2, main="Intensity distribution of Normalized data", cex.axis=0.6, col=sampleColor, names=sampleNames)
plot(clust.euclid.average, labels=sampleNames, main="Hierarchical clustering of NormData", cex=0.7,  hang=-1)
plotPCA(exprs(eset), labels=sampleNames, dataDesc="selected samples", colors=sampleColor,formapunts=c(rep(16,4),rep(17,4)),myCex=0.6)
```




